# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/extensions/webrequest/StreamFilterParent.cpp
# Commit: c58710d55a77
# Full Hash: c58710d55a771f83a36c1c7504d2edbeb6c1d1b5
# Author: Randell Jesup <rjesup@mozilla.com>
# Date: 2024-03-20 04:48:34
# Regressor Bug: 1356686
# File Overlap Count: 1
# Description:
#   Bug 1356686 - Do decompression off main thread in content r=necko-reviewers,valentin,extension-reviewers,robwu
#   
#   Content processes will now always retarget delivery of OnDataAvailable for Http
#   channels off the main thread. Consumers that were previously redirecting
#   off-main thread are not affected and their retargeting will stick, but any
# ==============================================================================

diff -r 969c8ce344d6 -r c58710d55a77 toolkit/components/extensions/webrequest/StreamFilterParent.cpp
--- a/toolkit/components/extensions/webrequest/StreamFilterParent.cpp	Tue Mar 19 17:27:22 2024 +0000
+++ b/toolkit/components/extensions/webrequest/StreamFilterParent.cpp	Tue Mar 19 17:27:22 2024 +0000
@@ -230,7 +230,7 @@
   if (trsl) {
     return trsl->CheckListenerChain();
   }
-  return NS_ERROR_FAILURE;
+  return NS_ERROR_NO_INTERFACE;
 }
 
 NS_IMETHODIMP